src/main/java/org/cyclonedx/util/ExtensibleTypesSerializer.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.cyclonedx.util;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator;
+import org.cyclonedx.model.Attribute;
+import org.cyclonedx.model.ExtensibleType;
+
+public class ExtensibleTypesSerializer extends StdSerializer<List<ExtensibleType>>
+{
+  private final String XMLNS = "xmlns";
+
+  private final String DEFAULT_VALID_NAMESPACE = "http://www.w3.org/1999/xhtml";
+
+  public ExtensibleTypesSerializer() {
+    this(null);
+  }
+
+  public ExtensibleTypesSerializer(final Class t) {
+    super(t);
+  }
+
+  @Override
+  public void serialize(
+      final List<ExtensibleType> extList,
+      final JsonGenerator generator,
+      final SerializerProvider provider)
+      throws IOException
+  {
+    final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) generator;
+    final XMLStreamWriter staxWriter = toXmlGenerator.getStaxWriter();
+    try {
+      if (extList != null && !extList.isEmpty()) {
+        for (ExtensibleType ext : extList) {
+          if (ext.getAttributes() != null && !ext.getAttributes().isEmpty()) {
+            Attribute xmlNS = ext.getAttributes().stream()
+                .filter(a -> a.getKey().contains(XMLNS))
+                .findAny()
+                .orElse(null);
+
+            if (xmlNS != null) {
+              staxWriter.writeStartElement(ext.getNamespace(), ext.getName(), xmlNS.getValue());
+              if (ext.getAttributes() != null && !ext.getAttributes().isEmpty()) {
+                for (Attribute attr : ext.getAttributes()) {
+                  staxWriter.writeAttribute(attr.getKey(), attr.getValue());
+                }
+              }
+            }
+          } else {
+            staxWriter.writeStartElement(ext.getNamespace(), ext.getName(), DEFAULT_VALID_NAMESPACE);
+          }
+
+          if (ext.getExtensibleTypes() != null && !ext.getExtensibleTypes().isEmpty()) {
+            serialize(ext.getExtensibleTypes(), generator, provider);
+          }
+          if (ext.getValue() != null) {
+            staxWriter.writeCharacters(ext.getValue());
+          }
+          staxWriter.writeEndElement();
+        }
+      }
+    }
+    catch (XMLStreamException e) {
+      e.printStackTrace();
```

Comment:
I think it's better to throw the exception

Reply:
Since this is an override, I'll have to throw it as `IOException`, do you see any issue with that?

Reply:
In: b53fa28